app/controllers/bookmarks_controller.rb

```diff
@@ -0,0 +1,59 @@
+class BookmarksController < ApplicationController
+  def show
+    @bookmark = Bookmark.find(params[:id])
+  end
+
+  def new
+    @topic = Topic.find(params[:topic_id])
+    @bookmark = Bookmark.new
+  end
+
+  def edit
+    topic = Topic.find(params[:topic_id])
+    @bookmark = topic.bookmarks.find(params[:id])
+  end
+
+  def update
+    topic = Topic.find(params[:topic_id])
+    @bookmark = topic.bookmarks.find(params[:id])
+    if @bookmark.update(bookmark_params)
+      flash[:notice] = 'bookmark was saved successfully.'
+      redirect_to topic
+    else
+      flash.now[:alert] = @bookmark.errors.full_messages.first
+      render :edit
+    end
+  end
+
+  def create
+    @topic = Topic.find(params[:topic_id])
+    @bookmark = @topic.bookmarks.build(bookmark_params)
+     #@bookmark.user = current_user
```

Comment:
because when it is in the application does not work i get back an error that says `NoMethodError in BookmarksController#create` `undefined method `user=' for #<Bookmark:0x007fed036f3d60>` should I just remove it. I meant to talk to you about it since I feel that I need something to assign the book mark to the user but it seems to already be doing that